### PR TITLE
Kafka components unflinkization next step

### PR DESCRIPTION
**Pull request by arkadius**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -827,7 +827,7 @@ lazy val baseEngineComponents = (project in engine("base/components")).
   settings(assemblySampleSettings("baseEngineComponents.jar"): _*).
   settings(
     name := "nussknacker-baseengine-components",
-  ).dependsOn(baseEngineApi)
+  ).dependsOn(baseEngineApi, avroUtil)
```
Comment (mproch) [inline build.sbt:830]: doesn't seem to be needed? :)

Comment (arkadius) [reply build.sbt:830]: it is part of longer story :D (forgot to setup Draft)

File: api/src/main/scala/pl/touk/nussknacker/engine/api/process/Source.scala [hunk 2]
```diff
@@ -43,12 +47,18 @@ trait SourceFactory[+T] extends Serializable with Component
 
 object SourceFactory {
 
-  def noParam[T: ClassTag](source: Source[T]): SourceFactory[T] =
-    new NoParamSourceFactory[T](source)
+  def noParam[T: ClassTag](source: Source[T], inputType: TypingResult): SourceFactory[T] =
```
[the remark this answers is not in the document]
Comment (arkadius) [reply api/src/main/scala/pl/touk/nussknacker/engine/api/process/Source.scala]: good point, wasn't before my change, but I can clean it

**Review by mproch — APPROVED** [review]
:+1:

**Pull request merged**